### ISSUE TITLE
stray "not covered" diagnostic in JSON output

### ISSUE THREAD

**Issue by ehuss** [rust-lang/rust#64769]
Since #64271, the E0005 error has begun emitting a strange "not covered" span in the JSON output. This does not appear in the normal "human" rendered output.

Sample code:
```rust
#![allow(unused)]

fn main() {
    let x = Some(1);
    let Some(y) = x;
}
```

With `--error-format=json`, this includes the following span which looks broken to me:

```javascript
   {
    "byte_end": 0,
    "byte_start": 0,
    "column_end": 1,
    "column_start": 1,
    "expansion": null,
    "file_name": "src/main.rs",
    "is_primary": false,
    "label": "not covered",
    "line_end": 1,
    "line_start": 1,
    "suggested_replacement": null,
    "suggestion_applicability": null,
    "text": [
      {
        "highlight_end": 1,
        "highlight_start": 1,
        "text": "#![allow(unused)]"
      }
    ]
  }
```

For tools that consume JSON output, this results in a strange message being displayed.

cc @Centril

**Comment by Centril** [comment]
Can you check if this reproduced for `match` before the PR?

Seems to be a cross crate issue?

```rust
#![allow(unused)]

enum O<T> { Some(T), None, }

use O::*;

fn main() {
    let x = Some(1);
    let Some(y) = x;
}
```
shows `not covered`.

Relevant code: https://github.com/Centril/rust/blob/6ef275e6c3cb1384ec78128eceeb4963ff788dca/src/librustc_mir/hair/pattern/check_match.rs#L502

cc @estebank

**Comment by estebank** [comment]
It's likely pointing at a `DUMMY_SP`, consistent with 

```
    "byte_end": 0,
    "byte_start": 0,
    "column_end": 1,
    "column_start": 1,
```

This is probably coming from desugaring. We can add a test around the `span_label` for this case, but we probably just do the same thing we do in the visual output and ignore `DUMMY_SP`s.

**Comment by ehuss** [comment]
Oh, I see. Yea, it works correctly for your example.  I guess it is trying to point to the definition of `Option`, but it is a dummy span.

**Comment by estebank** [comment]
How critical is this? Namely, how quickly do we need a fix? Fixing for this one test is a small pr, fixing the general case will be more arduous, I think.

**Comment by ehuss** [comment]
For me, not critical at all.  It would be nice to eventually get it fixed, but there is absolutely no rush.

**Comment by steveklabnik** [comment]
Triage: I believe this is fixed? I'm seeing `  "label":"expected one of `!` or `::`",` now.

**Comment by ehuss** [comment]
Thanks for checking @steveklabnik, I think the dummy span issue was resolved by #70199.
